mods/seasonal/moves.js

```diff
@@ -1848,42 +1848,134 @@ let BattleMovedex = {
+		pp: 15,
+		priority: 0,
+		flags: {snatch: 1, mirror: 1},
 		onPrepareHit: function (target, source) {
 			this.attrLastMove('[still]');
+			this.add('-anim', source, "Swallow", source);
 		},
-		// Balancing Reasons
-		onTry: function (source, target) {
-			if (target.transformed || source.transformed) {
-				this.add('-fail', source);
-				return null;
-			}
-			this.attrLastMove('[still]');
-			this.add('-anim', source, "Memento", target);
-			this.add('-anim', target, "Parabolic Charge", target);
-		},
-		onHit: function (target, pokemon) {
-			if (!pokemon.transformInto(target, pokemon)) {
-				return false;
-			}
-		},
-		secondary: {
-			chance: 100,
-			volatileStatus: 'flinch',
+		volatileStatus: 'purplepills',
+		effect: {
+			noCopy: true,
+			onStart: function (pokemon) {
+				this.add('-start', pokemon, 'Purple Pills', '[silent]');
+				this.add('-message', `${pokemon.name} downed some pills!`);
+				const allTypes = ['Normal', 'Fire', 'Fighting', 'Water', 'Flying', 'Grass', 'Poison', 'Electric', 'Ground', 'Psychic', 'Rock', 'Ice', 'Bug', 'Dragon', 'Ghost', 'Dark', 'Steel', 'Fairy'];
+				let type1 = allTypes[this.random(18)];
+				let type2 = allTypes[this.random(18)];
+				if (type1 === type2) {
+					pokemon.types = [type1];
+					this.add('-start', pokemon, 'typechange', `${type1}`);
+				} else {
+					pokemon.types = [type1, type2];
+					this.add('-start', pokemon, 'typechange', `${type1}/${type2}`);
+				}
+				// @ts-ignoretrack percentages to keep purple pills from resetting pp
+				pokemon.ppPercentages = pokemon.moveSlots.slice().map(m => {
+					return m.pp / m.maxpp;
+				});
+				// 3 moves with different requirements so no for statement
+				let newMovep = [];
+				let randMoves = [];
+				for (let i in exports.BattleMovedex) {
+					let move = exports.BattleMovedex[i];
+					if (i !== move.id) continue;
+					if (move.isZ || move.isNonstandard || !move.isViable) continue;
+					if (move.category !== 'Special') continue;
+					if (move.type !== type1) continue;
+					randMoves.push(move);
+				}
+				newMovep.push(toId(randMoves[this.random(randMoves.length)]));
+				randMoves = [];
+				for (let i in exports.BattleMovedex) {
+					let move = exports.BattleMovedex[i];
+					if (i !== move.id) continue;
+					if (move.isZ || move.isNonstandard || !move.isViable) continue;
+					if (move.category !== 'Special') continue;
+					if (move.type !== type2) continue;
+					randMoves.push(move);
+				}
+				newMovep.push(toId(randMoves[this.random(randMoves.length)]));
+				randMoves = [];
+				for (let i in exports.BattleMovedex) {
+					let move = exports.BattleMovedex[i];
+					if (i !== move.id) continue;
+					if (move.isZ || move.isNonstandard || !move.isViable) continue;
+					if (move.category !== 'Status') continue;
+					if (move.type !== type1 && move.type !== type2) continue;
+					randMoves.push(move);
+				}
+				newMovep.push(toId(randMoves[this.random(randMoves.length)]));
```

Comment:
Only loop once. Looping through the entire battlemovedex isnt a small thing, and should not be repeated.

To make this work, use something like this:
```javascript
if (move.category === 'Physical') continue;
if (!pokemon.types.includes(move.type) continue;
randMoves.push(move);
// when picking what moves to add to the movepool, simply track if you have a status move yet, if no and movepool >= 2, skip special, if yes, skip status.
```